reconcile/utils/state.py

```diff
@@ -59,99 +60,179 @@ def init_state(
-            bucket_name=state_bucket_name,
-            account_name=state_bucket_account_name,
-            accounts=aws_accounts,
-        )
+    s3_settings = aquire_state_settings(secret_reader)
 
-    # ... otherwise have a look if state settings are present in app-interface-settings-1.yml
-    state_settings = get_app_interface_state_settings()
-    if not state_settings:
-        raise StateInaccessibleException(
-            "app-interface state must be configured in order to use stateful integrations. "
-            "use one of the following options to provide state config: "
-            "* env vars APP_INTERFACE_STATE_BUCKET and APP_INTERFACE_STATE_BUCKET_ACCOUNT "
-            "* state settings in app-interface-settings-1.yml"
-        )
+    return State(
+        integration=integration,
+        bucket=s3_settings.bucket,
+        client=s3_settings.build_client(),
+    )
+
+
+class S3StateConfiguration(BaseModel):
+    bucket: str
+    region: str
+
+    @abstractmethod
+    def build_client(self) -> S3Client:
+        pass
 
-    if isinstance(state_settings, AppInterfaceStateConfigurationS3V1):
-        return _init_state_from_settings(
-            integration=integration,
-            secret_reader=secret_reader,
-            state_settings=state_settings,
+
+class S3CredsBasedStateConfiguration(S3StateConfiguration):
+    access_key_id: str
+    secret_access_key: str
+
+    def build_client(self) -> S3Client:
+        session = boto3.Session(
+            aws_access_key_id=self.access_key_id,
+            aws_secret_access_key=self.secret_access_key,
+            region_name=self.region,
         )
+        return session.client("s3")
 
-    raise StateInaccessibleException(
-        f"app-interface-settings-1.yml state provider {state_settings.provider} is not supported."
-    )
 
+class S3ProfileBasedStateConfiguration(S3StateConfiguration):
+    profile: str
 
-def _init_state_from_settings(
-    integration: str,
-    secret_reader: SecretReaderBase,
-    state_settings: AppInterfaceStateConfigurationS3V1,
-) -> "State":
+    def build_client(self) -> S3Client:
+        session = boto3.Session(profile_name=self.profile, region_name=self.region)
+        return session.client("s3")
+
+
+def aquire_state_settings(secret_reader: SecretReaderBase) -> S3StateConfiguration:
     """
-    Initializes a state object from the app-interface settings.
+    Finds the settings for the app-interface state provider in the following order:
 
-    :raises StateInaccessibleException: if the bucket is missing
-    or not accessible
+    * env variables pointing to a bucket and an AWS profile
+    * env variables pointing to static credentials
+    * env variables pointing to a bucket and a vault secret for creds
+    * env variables pointing to a bucket and an AWS account from app-interface for creds
+    * state settings in app-interface-settings-1.yml
+
+    If no settings can be found, a StateInaccessibleException is raised.
```

Comment:
Can we use default boto3 credentials behaviour to let it auto load? https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html

Since this config is a global one, we can inject secrets from vault to env vars, then let boto3 auto loader do the work.

Reply:
i would like us to stay in full control for now how we authenticate.
but i'm sure we need to reconsider anyways when we adopt more pieces of q-r to profile, STS etc. 
lets come back to this then.